src/VideoNode.cpp

```diff
@@ -73,6 +73,57 @@ void VideoNode::setChains(QList<QSharedPointer<Chain>> chains) {
     }
 }
 
+bool VideoNode::frozenInput() {
+    QMutexLocker locker(&m_stateLock);
+    return m_frozenInput;
+}
+
+void VideoNode::setFrozenInput(bool value) {
```

Comment:
This is OK for now, but it really should be frozen "edges". I think this can be implemented within `Model` and end up not really modifying `VideoNode`